datev_export_dtvf/models/datev_export_dtvf.py

```diff
@@ -0,0 +1,346 @@
+                    "state": "done",
+                    "date_generated": fields.Datetime.now(),
+                }
+            )
+
+    def _get_data_transaction(self, move):
+        # split move into single transactions from one account to another
+        move_line2amount = {
+            move_line: move_line.credit or move_line.debit
+            for move_line in move.line_ids
+            if not move_line.display_type and (move_line.debit or move_line.credit)
+        }
+        currency = move.currency_id or move.company_id.currency_id
+        code_length = move.company_id.datev_account_code_length
+        while move_line2amount:
+            move_line = min(move_line2amount, key=move_line2amount.get)
+            amount = move_line2amount.pop(move_line)
+            move_line2 = move_line
+            for move_line2 in move_line2amount:
+                if (
+                    move_line.debit
+                    and not move_line2.debit
+                    or move_line.credit
+                    and not move_line2.credit
+                ):
+                    move_line2amount[move_line2] = currency.round(
+                        move_line2amount[move_line2] - amount
+                    )
+                    if currency.is_zero(move_line2amount[move_line2]):
+                        move_line2amount.pop(move_line2)
+                    break
+            if move_line.account_id.internal_type not in (
+                "receivable",
+                "payable",
+            ) and move_line2.account_id.internal_type in ("receivable", "payable"):
+                move_line, move_line2 = move_line2, move_line
+            if move_line.account_id.datev_export_nonautomatic:
+                move_line, move_line2 = move_line2, move_line
+            account_number = move_line.account_id.code[-code_length:]
+            offset_account_number = move_line2.account_id.code[-code_length:]
+            if self.company_id.datev_partner_numbering in ("ee", "sequence"):
+                for ml in (move_line, move_line2):
+                    number = (
+                        account_number if ml == move_line else offset_account_number
+                    )
+                    number_type = (
+                        "customer"
+                        if ml.account_id.internal_type == "receivable"
+                        else (
+                            "supplier"
+                            if ml.account_id.internal_type == "payable"
+                            else None
+                        )
+                    )
+                    if number_type and ml.partner_id:
+                        number = self._get_partner_number(
+                            ml.partner_id, number_type, True
+                        )
+                    if ml == move_line:
+                        account_number = number
+                    else:
+                        offset_account_number = number
+            data = {
```

Comment:
```python
if account_number == offset_account_number:
    continue
```

We checked the data again and the lines we could not validate where bookings between the same accounts. Because nothing changes with these we don't need these lines in the CSV.